usageReports/ebs/prepare_response.go

```diff
@@ -0,0 +1,174 @@
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package ebs
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/trackit/jsonlog"
+	"gopkg.in/olivere/elastic.v5"
+
+	"github.com/trackit/trackit-server/aws/usageReports"
+	"github.com/trackit/trackit-server/aws/usageReports/ebs"
+	"github.com/trackit/trackit-server/errors"
+)
+
+type (
+
+	// Structure that allow to parse ES response for costs
+	ResponseCost struct {
+		Accounts struct {
+			Buckets []struct {
+				Key       string `json:"key"`
+				Snapshots struct {
+					Buckets []struct {
+						Key  string `json:"key"`
+						Cost struct {
+							Value float64 `json:"value"`
+						} `json:"cost"`
+					} `json:"buckets"`
+				} `json:"snapshots"`
+			} `json:"buckets"`
+		} `json:"accounts"`
+	}
+
+	// Structure that allow to parse ES response for EBS Monthly snapshots
+	ResponseEbsMonthly struct {
+		Accounts struct {
+			Buckets []struct {
+				Snapshots struct {
+					Hits struct {
+						Hits []struct {
+							Snapshot ebs.SnapshotReport `json:"_source"`
+						} `json:"hits"`
+					} `json:"hits"`
+				} `json:"snapshots"`
+			} `json:"buckets"`
+		} `json:"accounts"`
+	}
+
+	// Structure that allow to parse ES response for EBS Daily snapshots
+	ResponseEbsDaily struct {
+		Accounts struct {
+			Buckets []struct {
+				Dates struct {
+					Buckets []struct {
+						Time      string `json:"key_as_string"`
+						Snapshots struct {
+							Hits struct {
+								Hits []struct {
+									Snapshot ebs.SnapshotReport `json:"_source"`
+								} `json:"hits"`
+							} `json:"hits"`
+						} `json:"snapshots"`
+					} `json:"buckets"`
+				} `json:"dates"`
+			} `json:"buckets"`
+		} `json:"accounts"`
+	}
+
+	// SnapshotReport has all the information of an EBS snapshot report
+	SnapshotReport struct {
+		utils.ReportBase
+		Snapshot Snapshot `json:"snapshot"`
+	}
+
+	// Snapshot contains the information of an EBS snapshot
+	Snapshot struct {
+		ebs.SnapshotBase
+		Tags   map[string]string  `json:"tags"`
+		Volume Volume             `json:"volume"`
+		Cost  float64             `json:"cost"`
+	}
+
+	// Volume contains information about EBS volumes
+	Volume struct {
+		Id    string  `json:"id"`
+		Size  int64   `json:"size"`
+	}
+)
+
+func getEbsSnapshotReportResponse(oldSnapshot ebs.SnapshotReport) SnapshotReport {
+	tags := make(map[string]string, 0)
+	for _, tag := range oldSnapshot.Snapshot.Tags {
+		tags[tag.Key] = tag.Value
+	}
+	newSnapshot := SnapshotReport{
+		ReportBase: oldSnapshot.ReportBase,
+		Snapshot: Snapshot{
+			SnapshotBase: oldSnapshot.Snapshot.SnapshotBase,
+			Tags:         tags,
+			Cost:        oldSnapshot.Snapshot.Cost,
+			Volume: Volume{
+				Id:   oldSnapshot.Snapshot.Volume.Id,
+				Size: oldSnapshot.Snapshot.Volume.Size,
+			},
+		},
```

Comment:
Make sure to run `gofmt` to have the right alignment for data in these structures

Reply:
Fixed